### PR TITLE
Add an ability to inject an arbitrary set of headers to KV requests

### PR DESCRIPTION
**Pull request by vlovich**
We want to start injecting some headers in the multi-tenant runtime.

### CODE REVIEW

**Comment by vlovich** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by vlovich** [comment]
recheck

File: src/workerd/api/kv.h [hunk 1]
```diff
@@ -13,9 +13,12 @@ class KvNamespace: public jsg::Object {
   // A capability to a KV namespace.
 
 public:
-  explicit KvNamespace(uint subrequestChannel): subrequestChannel(subrequestChannel) {}
+  explicit KvNamespace(uint subrequestChannel,
+      kj::Array<kj::Tuple<kj::String, kj::String>> additionalHeaders = {})
```
Comment (bretthoerner) [inline src/workerd/api/kv.h:17]: super nit: Kinda weird to swap the swap the parameter and field order for no reason, and we try to stay away from tuples (in favor of structs with named fields), but I don't know how someone could confuse a tuple of headers... so :shrug:

**Review by bretthoerner — APPROVED** [review]

File: src/workerd/api/kv.h (hunk 1 above)
Comment (kentonv) [reply src/workerd/api/kv.h:17]: Yeah please don't use `kj::Tuple` unless you're doing variadic template metaprogramming. A struct with two members is much more readable.

Comment (vlovich) [reply src/workerd/api/kv.h:17]: Got it. The nit reason I swapped the field order is that putting a uint first introduces padding & I can't have a first parameter be default. I can put additional headers first if I get rid of it. Let me know.

Comment (vlovich) [reply src/workerd/api/kv.h]: Went with option 2 - rearrange it so that the array comes first & updated `workerd-api` to reflect that.

**Comment by bretthoerner** [comment]
LGTM

**Review by a-robinson — APPROVED** [review]

**Pull request merged**